dotnet/samples/TeamsChefBot/Program.cs

```diff
@@ -0,0 +1,93 @@
+using TeamsChefBot;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.TeamsAI;
+using Microsoft.TeamsAI.AI.Planner;
+using Microsoft.TeamsAI.State;
+using Microsoft.TeamsAI.AI.Prompt;
+using Microsoft.TeamsAI.AI;
+using Microsoft.TeamsFx.Conversation;
+using Microsoft.TeamsAI.AI.Moderator;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddHttpClient("WebClient", client => client.Timeout = TimeSpan.FromSeconds(600));
+builder.Services.AddHttpContextAccessor();
+
+// Prepare Configuration for ConfigurationBotFrameworkAuthentication
+var config = builder.Configuration.Get<ConfigOptions>();
+builder.Configuration["MicrosoftAppType"] = "MultiTenant";
+builder.Configuration["MicrosoftAppId"] = config.BOT_ID;
+builder.Configuration["MicrosoftAppPassword"] = config.BOT_PASSWORD;
+
+// Create the Bot Framework Authentication to be used with the Bot Adapter.
+builder.Services.AddSingleton<BotFrameworkAuthentication, ConfigurationBotFrameworkAuthentication>();
+
+// Create the Cloud Adapter with error handling enabled.
+// Note: some classes expect a BotAdapter and some expect a BotFrameworkHttpAdapter, so
+// register the same adapter instance for all types.
+builder.Services.AddSingleton<CloudAdapter, AdapterWithErrorHandler>();
+builder.Services.AddSingleton<IBotFrameworkHttpAdapter>(sp => sp.GetService<CloudAdapter>());
+builder.Services.AddSingleton<BotAdapter>(sp => sp.GetService<CloudAdapter>());
+
+// Create the Conversation Bot.
+builder.Services.AddSingleton(sp =>
+{
+    ConversationOptions options = new ConversationOptions()
+    {
+        Adapter = sp.GetService<CloudAdapter>(),
+        Command = new CommandOptions()
+        {
+            Commands = new List<ITeamsCommandHandler>()
+        }
+    };
+    return new ConversationBot(options);
+});
+
+// Create the Application.
+builder.Services.AddSingleton<IBot, TeamsChefBotApplication>(sp =>
```

Comment:
Better to add transient for `IBot` to keep consistence with other dotnet bot samples.
But ok to add singleton for those thread-safe instance, e.g. storage, static options, ...

Reply:
Updated the style. Not sure if I fully catch your idea.

Reply:
Looks good